Add docs for using Stripe in test mode (#10153)
hackclub/hcb: dev-docs/development.md

## Credentials

> External contributors should provide credentials via a `.env.development` file [(view example)](.env.development.example). Note that if you're using the devcontainer (such as with GitHub Codespaces), you must rebuild the container after modifying the `.env.development` file to pull in the new variables.

Reviewer: ```suggestion
External contributors should provide credentials via a `.env.development` file [(view example)](.env.development.example). Developers using the `devcontainer` setup (eg. in GitHub Codespaces), will need to rebuild the container after modifying the `.env.development` file to pull in the new variables.
```
Hmm - this is slightly odd - does `bin/rails s` and rerunning it not also reload the credentials? I might be missing something here.

Author: From what I understood the only thing pulling in credentials from .env.development was the docker compose configuration:
https://github.com/hackclub/hcb/blob/6abe345efca463847b7a10bbe3fa52008bbcd8a5/.devcontainer/docker-compose.yml#L20

When I was first setting this up I tried re-running `bin/rails dev` and the environment variables didn't seem to update

> HCB relies on two services for the majority of it's financial features: Stripe and Column. See [the Stripe testing guide](./stripe_testing.md) for getting setup with Stripe in development. You can register for a Column account [here](https://dashboard.column.com/register); after their onboarding questions, select "Skip to Sandbox".

Reviewer: ```suggestion
HCB relies on two services for the majority of it's financial features: Stripe and Column. Follow [the Stripe testing guide](./stripe_testing.md) to setup Stripe. You can register for a Column account [here](https://dashboard.column.com/register); after their onboarding questions, select "Skip to Sandbox".
```
